### PR TITLE
testv2: Port cloud provider tests

### PR DESCRIPTION
**Pull request by xmudrii**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

This PR ports [the cloud provider tests from the old tests](https://github.com/kubermatic/kubeone/blob/4fd7276f5623ef4ad42e7371f106eea478268d02/test/e2e/cloudprovider/cloudprovider.go) to the new tests.

Those tests are needed to properly test CCM and CSI drivers.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: testv2/e2e/cloudprovider.go [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+
+func (c *cloudProviderTests) run(t *testing.T) error {
+	if err := c.createPVCWithStorage(t); err != nil {
+		return err
+	}
+	if err := c.exposePVC(t); err != nil {
+		return err
+	}
+
+	return c.cleanUp(t)
```
Comment (kron4eg) [inline testv2/e2e/cloudprovider.go:88]: cleanUp should be a defer function, or even better a [testing.T.Cleanup](https://pkg.go.dev/testing#T.Cleanup), and it should be registered before any cloud-resource-producing functions are run.

Comment (xmudrii) [reply testv2/e2e/cloudprovider.go]: This should be fixed now.

File: testv2/e2e/scenario_migrate_csi_ccm.go [hunk 2]
```diff
@@ -73,33 +73,27 @@ func (scenario *scenarioMigrateCSIAndCCM) Run(t *testing.T) {
 	k1Old := install.kubeone(t)
 	waitKubeOneNodesReady(t, k1Old)
 
+	client := dynamicClientRetriable(t, k1Old)
+	cpTests := newCloudProviderTests(client, scenario.infra.Provider())
+	defer cpTests.cleanUp(t)
```
Comment (kron4eg) [inline testv2/e2e/scenario_migrate_csi_ccm.go:78]: cpTests.runWithCleanup()

Comment (xmudrii) [reply testv2/e2e/scenario_migrate_csi_ccm.go:78]: This is intentional. I want to ensure that once the CCM/CSI migration is done, the pod with PVC is Running and LB is accessible. For that, I intentionally defer cleanup until the end.

File: testv2/e2e/cloudprovider.go [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+
+		return
+	}
+
+	c.createStatefulSetWithStorage(t)
+	c.exposeStatefulSet(t)
+}
+
+func (c *cloudProviderTests) runWithCleanup(t *testing.T) {
+	defer c.cleanUp(t)
```
Comment (kron4eg) [inline testv2/e2e/cloudprovider.go:84]: This can be t.Cleanup(), it defer the cleanup cross the calls.

File: testv2/e2e/cloudprovider.go [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+		t.Logf("Skipping cloud provider tests because cloud provider %q is not supported.", c.provider)
+
+		return
+	}
+
+	c.createStatefulSetWithStorage(t)
+	c.exposeStatefulSet(t)
+}
+
+func (c *cloudProviderTests) runWithCleanup(t *testing.T) {
```
Comment (kron4eg) [inline testv2/e2e/cloudprovider.go:83]: Why not cleanUp always inside run()? If we didn't created anything, cleanUp() would simply return after few 404 GETs.

Comment (xmudrii) [reply testv2/e2e/cloudprovider.go:83]: Please see https://github.com/kubermatic/kubeone/pull/2177#discussion_r940343385

File: testv2/e2e/scenario_migrate_csi_ccm.go (hunk 2 above)
Comment (kron4eg) [reply testv2/e2e/scenario_migrate_csi_ccm.go:78]: You could `t.Cleanup()` in that case. And `cpTests.run()` can do it for you ;)

Comment (kron4eg) [reply testv2/e2e/scenario_migrate_csi_ccm.go:78]: `t.Cleanup()` defers until the very end of the whole test!

Comment (xmudrii) [reply testv2/e2e/scenario_migrate_csi_ccm.go:78]: I want it to work in the following way:

* non-CCM/CSI migration tests: cleanup as soon as possible, i.e. before starting Sonobuoy tests (that's why `defer` is useful here)
* CCM/CSI migration tests: cleanup at the end, after making sure everything is running after the migration (note that we also don't have Sonobuoy tests for CCM/CSI migration, which is completely okay)

The first case is to save resources/money and leave space for other tests to run. Sonobuoy tests can take 10-20 minutes (or even longer if you run a bigger suite), so by cleaning up earlier, we let more tests run in parallel.

**Pull request merged**